compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -333,7 +333,8 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
   }
 
   private fun ModuleDescriptor.mangledNameSuffix(): String {
-    val name = name.asString()
+    // We replace - with _ to maintain interoperability with Dagger's expected generated identifiers
+    val name = name.asString().replace('-', '_')
```

Comment:
I'm curious, without this fix would the test below fail in CI (Windows in particular)?

Reply:
Good question! It fails without the fix on my machine, but I just opened https://github.com/square/anvil/pull/656 to verify we see a failure in CI as well.